implementations/rust/ockam/ockam_command/src/node/show.rs

```diff
@@ -52,10 +52,51 @@ async fn run_impl(
     ctx: Context,
     (opts, cmd): (CommandGlobalOpts, ShowCommand),
 ) -> miette::Result<()> {
-    let node_name = get_node_name(&opts.state, &cmd.node_name);
+    if let Some(node_name) = cmd.node_name {
+        show_node(&opts, &ctx, &node_name).await?;
+    } else if !opts.terminal.can_ask_for_user_input() {
+        show_node(&opts, &ctx, &get_default_node_name(&opts.state)).await?;
+    } else {
+        let node_names = opts.state.nodes.list_items_names()?;
+        if node_names.is_empty() {
+            opts.terminal
+                .stdout()
+                .plain("There are no nodes to show")
+                .write_line()?;
+            return Ok(());
+        }
+
+        let selected_node_names = opts.terminal.select_multiple(
+            "Select one or more nodes that you want to show".to_string(),
+            node_names,
+        );
+
+        if selected_node_names.is_empty() {
+            opts.terminal
+                .stdout()
+                .plain("No nodes selected to show")
+                .write_line()?;
+            return Ok(());
+        }
+
+        if opts.terminal.confirm_interactively(format!(
+            "Would you like to show these items : {:?}?",
+            selected_node_names
+        )) {
+            for selected_node_name in selected_node_names {
+                show_node(&opts, &ctx, &selected_node_name).await?;
```

Comment:
You should use the `NodeListOutput` struct to print the list of nodes, which shows less information per item. Perhaps the `show_node` function extraction is not necessary if you do this.

Reply:
Hi @adrianbenavides, I have a question the result between `node list` and `node show` is quite different. 

`node list`
![image](https://github.com/build-trust/ockam/assets/3379504/b1a3639d-2497-4703-9161-5ad847a5a776)

`node show`
![image](https://github.com/build-trust/ockam/assets/3379504/676371d2-5f49-4af6-8e88-1b1f1919823d)

I'd like to clarify whether it is expected to see the `node list` result when selecting multiple nodes in `node show`?
If so, should I directly use the function and struct from list.rs in show.rs?

Reply:
Sorry for the delay in answering this question. Yeah, I think we should use the `list` format when showing multiple nodes. Good catch!

Reply:
Hi @adrianbenavides, to print `node list` format, I made some `node::list` functions public. Please help review again. Thanks!